### DISCUSSION TITLE
Formatter in neovim

### DISCUSSION THREAD

**Discussion by DashieTM** [nvarner/typst-lsp#306] [Q&A] [answered]
First of all, big thanks for the awesome LSP!
Small question about typstfmt integration:
Is there something specific I need to configure besides setting the experimental flag?
Other LSPs like rust-analyzer, gopls, etc. work as expected.
( lazyvim setup with nvim-lspconfig and mason )
```
typst_lsp = {
    settings = {
        experimentalFormatterMode = "on",
    },
},
```

**Comment by KillTheMule** [comment]
If you use `typst_lsp` as shown above as an argument for `nvim_lsp.typst_lsp.setup`, it whould work. What problem are you seeing?

**Reply by DashieTM** [reply]
The LSP format request fails, with no matching servers according to the response.

My lsp config file is wayy to large to post here so: https://github.com/DashieTM/dotFiles/blob/main/nvim/lua/plugins/lsp.lua

My initial assumption was that I need to setup the formatter differently to other language servers.

**Reply by KillTheMule** [reply]
My guess is you need to lowercase your `On`, as in

```lua
        typst_lsp = {
          settings = {
            experimentalFormatterMode = "on",
            exportPdf = "onSave",
          },
```

**Reply by DashieTM** [reply]
Oh sorry, that is old, I am trying different things, had it on lowercase before, and also have it on lowercase now.

**Reply by KillTheMule** [reply]
Does `:LspInfo` tell you the lsp is attached? How exactly are you calling formatting? Your config is a tad to extensive for me to go through, so try

```vim
:lua vim.lsp.buf.format()
```

**Reply by DashieTM** [reply]
That is the command that gives me the response I mentioned above.
It is attached though.

I have a workaround for now, just forked null-ls and put typstfmt in there.

**Reply by KillTheMule** [reply, marked as answer]
Huh, I'm kinda having a dejavu, but ... it's working for me :) My config is much more simple though

```lua
nvim_lsp["typst_lsp"].setup{
  on_attach = on_attach,
  root_dir = function(fname) return util.path.dirname(fname) end,
  settings = {
    exportPdf = "never",
    experimentalFormatterMode = "on",
  },
}
```

**Reply by DashieTM** [reply]
Might have something to do with the lazyvim formatter, I guess it has downsides and upsides when using a distribution.
But thanks a lot for helping, I at least definitely know it's my config and not some special formatter thing. :D

**Comment by avonmoll** [comment]
Is there a way to specify a configuration for `typstfmt` or to respect its global configuration (set via `typstfmt.toml`)? I created a configuration and placed it in the appropriate folder (given by `typstfmt --get-global-config-path`). My config is respected if I manually use `typstfmt` but not when used by `typst-lsp`.

**Reply by avonmoll** [reply]
After some digging, I discovered that you can specify a `typstfmt` config by placing the config in a file called `typstfmt-config.toml` in the project root.

**Reply by KillTheMule** [reply]
Yeah the global config is not yet supported. I think when I implemented the config the option `--get-global-config-path` did not yet exist in `typstfmt`.

**Comment by GaetanLepage** [comment]
I face the same issue. My configuration is the same as [this one](https://github.com/nvarner/typst-lsp/discussions/306#discussioncomment-7073810).
However, when I run `:=vim.lsp.buf.get_active_clients()[1].server_capabilities`, I don't have `documentFormattingProvider = true`.
Because of that, when I run `:lua vim.lsp.buf.format()`, I get
```
[LSP] Format request failed, no matching language servers.
```

Note: I am running
```
Version: 0.13.0, commit unknown (Typst version 0.11.0)
```
installed through nix.
